distributed/worker_state_machine.py

```diff
@@ -1863,22 +1882,12 @@ def _transition_executing_rescheduled(
         Reschedule() Exception; it is not involved in work stealing.
         The task is always done.
         """
-        if self.validate:
-            # Notably, we're missing the third state in which a task can raise
-            # Reschedule(), which is "cancelled"
-            assert ts.state in ("executing", "long-running"), ts
-
-        self._release_resources(ts)
-        self.executing.discard(ts)
-        self.long_running.discard(ts)
-
         return merge_recs_instructions(
             ({}, [RescheduleMsg(key=ts.key, stimulus_id=stimulus_id)]),
             # Note: this is not the same as recommending {ts: "released"} on the
             # previous line, as it would instead transition the task to cancelled - but
             # a task that raised the Reschedule() exception is finished!
```

Comment:
Is this comment still relevant? I don't really understand it and it seems to reference a line that no longer exists

Reply:
The comment is still valid.
`transition_executing_released` and `transition_resumed_released` contain `assert not ts.done`. I could of course change them with a special code path that is exclusive of the rescheduled state, but I thought that keeping the logic within `transition_*_rescheduled` was a lot more readable.

Reply:
follow-up: #6685